From 3ade5f5022cd472cc7bbbf1995f0cbcc89e06410 Mon Sep 17 00:00:00 2001
Message-Id: <patch-v6-03.11-3ade5f5022c-20230307T175532Z-avarab@gmail.com>
In-Reply-To: <cover-v6-00.11-00000000000-20230307T175532Z-avarab@gmail.com>
References: <cover-v5-00.10-00000000000-20230207T154000Z-avarab@gmail.com>
	<cover-v6-00.11-00000000000-20230307T175532Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Tue, 7 Mar 2023 11:55:34 +0100
Subject: [PATCH v6 03/11] config.h docs: git_config_get_string() never returns
 -1
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This documentation initially added in [1] has been incorrect since
[2], when we started invoking git_die_config() in cases where we'd
previously return -1 from this function.

As a subsequent commit will add a case where a related function will
return negative values, leaving this in place would be confusing.

1. 3c8687a73ee (add `config_set` API for caching config-like files, 2014-07-28)
2. 5a80e97c827 (config: add `git_die_config()` to the config-set API, 2014-08-07)

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 config.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config.h b/config.h
index 7606246531a..2eca084cb1c 100644
--- a/config.h
+++ b/config.h
@@ -547,8 +547,8 @@ void git_config_clear(void);
 
 /**
  * Allocates and copies the retrieved string into the `dest` parameter for
- * the configuration variable `key`; if NULL string is given, prints an
- * error message and returns -1. When the configuration variable `key` is
+ * the configuration variable `key`; dies with an error if NULL string is given.
+ * When the configuration variable `key` is
  * not found, returns 1 without touching `dest`.
  */
 int git_config_get_string(const char *key, char **dest);
-- 
2.40.0.rc1.1034.g5867a1b10c5

